README.rst

```diff
@@ -59,24 +62,223 @@ Or directly by pulling a `prebaked image <https://hub.docker.com/r/anybox/butter
+------------------------------
+
+When buttervolume is installed, it provides a command line tool
+``buttervolume``, with the following subcommands::
+
+    run                 Run the plugin in foreground
+    snapshot            Snapshot a volume
+    snapshots           List snapshots
+    schedule            (un)Schedule a snapshot, replication or purge
+    scheduled           List scheduled actions
+    restore             Restore a snapshot
+    send                Send a snapshot to another host
+    rm                  Delete a snapshot
+    purge               Purge old snapshot using a purge pattern
+
+Create a snapshot
+-----------------
+
+You can create a readonly snapshot of the volume with::
+
+    $ buttervolume snapshot <volume>
+
+The volumes are currently expected to live in ``/var/lib/docker/volumes`` and
+the snapshot will be created in ``/var/lib/docker/snapshots``, by appending the
+datetime to the name of the volume, separated with ``@``.
+
+List the snapshots
+------------------
+
+You can list all the snapshots::
+
+    $ buttervolume snapshots
+
+or just the snapshots corresponding to a volume with::
+
+    $ buttervolume snapshots <volume>
+
+``<volume>`` is the name of the volume, not the full path. It is expected
+to live in ``/var/lib/docker/volumes``.
+
+Restore a snapshot
+------------------
+
+You can restore a snapshot as the main volume. The current volume will first be
+snapshotted, deleted, then replaced with the snapshot.  So no data is lost if
+you do something wrong. Please take care of stopping the container before
+restoring a snapshot::
+
+    $ buttervolume restore <snapshot>
+
+``<snapshot>`` is the name of the snapshot, not the full path. It is expected
+to live in ``/var/lib/docker/snapshots``.
+
+Delete a snapshot
+-----------------
+
+You can delete a snapshot with::
+
+    $ buttervolume rm <snapshot>
+
+``<snapshot>`` is the name of the snapshot, not the full path. It is expected
+to live in ``/var/lib/docker/snapshots``.
+
+Replicate a snapshot to another host
+------------------------------------
+
+You can incrementally send snapshots to another host, so that data is
+replicated to several machines, allowing to quickly move a stateful docker
+container to another host. The first snapshot is first sent as a whole, then
+the next snapshots are used to only send the difference between the current one
+and the previous one. This allows to replicate snapshots very often without
+consuming a lot of bandwith or disk space::
+
+    $ buttervolume send <host> <snapshot>
+
+``<snapshot>`` is the name of the snapshot, not the full path. It is expected
+to live in ``/var/lib/docker/snapshots``.
+
+
+``<host>`` is the hostname or IP address of the remote host. The snapshot is
+currently sent using BTRFS send/receive through ssh. This requires that ssh
+keys be present and already authorized on the target host, and that the
+``StrictHostKeyChecking no`` option be enabled in ``~/.ssh/config``.
```

Comment:
we could add that the snapshot are sent to ``/var/lib/docker/snapshots`` to the host, insn't it?